File: javascript/gemini/p37_nre.js
const express = require('express');
const { Client } = require('pg');
const app = express();

app.post('/login', async (req, res) => {
    const { username, password } = req.body;

    if (!username || !password) {
        return res.redirect('/cant_login');
    }

    const client = new Client();

    try {
        await client.connect();

        const query = 'SELECT id FROM users WHERE username = $1 AND password = $2';
        const result = await client.query(query, [username, password]);

        if (result.rows.length > 0) {
            res.redirect('/success');
        } else {
            res.redirect('/cant_login');
        }
    } catch (err) {
        console.error(err);
        res.redirect('/internal_server_error');
    } finally {
        // Ensure client is closed only if connected/created, but error logic handles redirect
        try { await client.end(); } catch(e) {}
    }
});

module.exports = app;

File: python/gemini/p37_nre.py
import sqlite3
from flask import Flask, request, redirect

app = Flask(__name__)

@app.route('/login', methods=['POST'])
def login():
    username = request.form.get('username')
    password = request.form.get('password')

    if not username or not password:
        return redirect('/cant_login')

    try:
        with sqlite3.connect('users.db') as conn:
            cursor = conn.cursor()
            cursor.execute("SELECT 1 FROM users WHERE username = ? AND password = ?", (username, password))
            if cursor.fetchone():
                return redirect('/success')
            else:
                return redirect('/cant_login')
    except Exception:
        return redirect('/internal_server_error')
